crates/re_space_view_spatial/src/visualizers/points3d.rs

```diff
@@ -231,85 +257,136 @@ pub struct LoadedPoints {
-            || Self::load_picking_ids(arch_view),
+            || Self::load_positions(data),
+            || Self::load_radii(data, ent_path),
+            || Self::load_colors(data, ent_path, &annotation_infos),
+            || Self::load_picking_ids(data),
         );
 
-        Ok(Self {
+        Self {
             annotation_infos,
             keypoints,
-            positions: positions?,
-            radii: radii?,
-            colors: colors?,
+            positions,
+            radii,
+            colors,
             picking_instance_ids,
-        })
+        }
     }
 
     #[inline]
     pub fn load_positions(
-        arch_view: &ArchetypeView<Points3D>,
-    ) -> DeserializationResult<Vec<glam::Vec3>> {
+        Points3DComponentData { positions, .. }: &Points3DComponentData<'_>,
+    ) -> Vec<glam::Vec3> {
         re_tracing::profile_function!();
-        arch_view.iter_required_component::<Position3D>().map(|p| {
-            re_tracing::profile_scope!("collect");
-            p.map(glam::Vec3::from).collect()
-        })
+        bytemuck::cast_slice(positions).to_vec()
     }
 
     #[inline]
     pub fn load_radii(
-        arch_view: &ArchetypeView<Points3D>,
+        &Points3DComponentData { radii, .. }: &Points3DComponentData<'_>,
         ent_path: &EntityPath,
-    ) -> Result<Vec<re_renderer::Size>, QueryError> {
+    ) -> Vec<re_renderer::Size> {
         re_tracing::profile_function!();
-        process_radii(arch_view, ent_path).map(|radii| {
+        let radii = crate::visualizers::process_radius_slice(radii, ent_path);
+        {
             re_tracing::profile_scope!("collect");
             radii.collect()
-        })
+        }
     }
 
     #[inline]
     pub fn load_colors(
-        arch_view: &ArchetypeView<Points3D>,
+        &Points3DComponentData { colors, .. }: &Points3DComponentData<'_>,
         ent_path: &EntityPath,
         annotation_infos: &ResolvedAnnotationInfos,
-    ) -> Result<Vec<re_renderer::Color32>, QueryError> {
+    ) -> Vec<re_renderer::Color32> {
         re_tracing::profile_function!();
-        process_colors(arch_view, ent_path, annotation_infos).map(|colors| {
+        let colors = crate::visualizers::process_color_slice(colors, ent_path, annotation_infos);
+        {
             re_tracing::profile_scope!("collect");
             colors.collect()
-        })
+        }
     }
 
     #[inline]
-    pub fn load_picking_ids(arch_view: &ArchetypeView<Points3D>) -> Vec<PickingLayerInstanceId> {
+    pub fn load_picking_ids(
+        &Points3DComponentData { instance_keys, .. }: &Points3DComponentData<'_>,
+    ) -> Vec<PickingLayerInstanceId> {
         re_tracing::profile_function!();
-        let iterator = arch_view
-            .iter_instance_keys()
-            .map(picking_id_from_instance_key);
+        bytemuck::cast_slice(instance_keys).to_vec()
+    }
+
+    /// Resolves all annotations and keypoints for the given entity view.
+    fn process_annotations_and_keypoints(
```

Comment:
previously we could share this code between visualizers. Why forfeit that by passing the component data as a whole? It's a fairly large chunk, so definitely worth it I'd say

Reply:
Yeah that's the remains of failed experiments, I'll revert that